google/cloud/pubsub/integration_tests/subscriber_integration_test.cc

```diff
@@ -523,6 +523,32 @@ TEST_F(SubscriberIntegrationTest, ExactlyOnce) {
   EXPECT_STATUS_OK(result.get());
 }
 
+TEST_F(SubscriberIntegrationTest, BlockingPull) {
+  auto publisher = Publisher(MakePublisherConnection(topic_));
+  auto subscriber =
+      Subscriber(MakeSubscriberConnection(exactly_once_subscription_));
+
+  std::set<std::string> ids;
+  for (auto const* data : {"message-0", "message-1", "message-2"}) {
+    auto response =
+        publisher.Publish(MessageBuilder{}.SetData(data).Build()).get();
+    EXPECT_STATUS_OK(response);
+    if (response) ids.insert(*std::move(response));
+  }
+  EXPECT_THAT(ids, Not(IsEmpty()));
+
+  auto const count = 2 * ids.size();
```

Comment:
Why do we do 2*N iterations for an exactly-once subscription?

Is it to make sure everything gets acked, in the failure case where we `continue`?

Reply:
Yes. For low values of "make sure".